go/database/mpt/io/live_test.go

```diff
@@ -95,7 +95,50 @@ func TestIO_ExportAndImportAsArchive(t *testing.T) {
 	}
 }
 
+func TestIO_ExportedDataIsDeterministic(t *testing.T) {
+	reference, _ := exportExampleState(t)
+	for i := 0; i < 10; i++ {
+		data, _ := exportExampleState(t)
+		if !bytes.Equal(data, reference) {
+			t.Fatalf("exported data is not deterministic")
+		}
+	}
+}
+
+func TestIO_ExportedDataDoesNotContainExtraCodes(t *testing.T) {
+	reference, referenceHash := exportExampleState(t)
+
+	// Modify the state by adding and removing code from an account.
+	// This temporary code should not be included in the resulting exported data.
+	modified, modifiedHash := exportExampleStateWithModification(t, func(s *mpt.MptState) {
+		addr1 := common.Address{1}
+		code, err := s.GetCode(addr1)
+		if err != nil {
+			t.Fatalf("failed to fetch code: %v", err)
+		}
+		modified := append(code, []byte("extra_code")...)
+		s.SetCode(addr1, modified)
+		s.SetCode(addr1, code)
+	})
+
+	// Check that the test indeed did not modify the state content.
+	if referenceHash != modifiedHash {
+		t.Fatalf("modified state is not equivalent to reference state")
+	}
+
+	// The extra code that was only temporary in the state should no be included
+	// in the exported data.
+	if !bytes.Equal(reference, modified) {
+		t.Fatalf("exported data contains extra codes")
+	}
+}
+
 func exportExampleState(t *testing.T) ([]byte, common.Hash) {
+	t.Helper()
+	return exportExampleStateWithModification(t, nil)
+}
+
+func exportExampleStateWithModification(t *testing.T, modify func(s *mpt.MptState)) ([]byte, common.Hash) {
```

Comment:
I think you can just call `exportExampleStateWithModification(t, nil)` and avoid creating this trivial func. Wdyt?

Reply:
The preexisting function `exportExampleState` has multiple prior uses in this file. Adding the additional `nil` would harm the readability of those cases, since it is not intuitively clear what the `nil` is for. Thus I favor this variant.